test/e2e/consumer_test.exs

```diff
@@ -48,16 +50,42 @@ defmodule E2E.ConsumerTest do
     %{connection: connection, pid: pid}
   end
 
-  setup context do
-    exit_fn = fn ->
-      purge_queue(context.connection, "railway:test:events")
-    end
+  @tag :e2e
+  test "successfully consumes a message in binary protobuf format", context do
+    # Make sure the queue is empty
+    Helpers.wait_for_true(@timeout, fn ->
+      assert 0 == queue_count("railway:test:events")
+    end)
```

Comment:
does it makes sense for this to maybe have a wrapper function like:

```suggestion
    wait_for_empty_queue("railway:test:events", @timeout) 
    
    def wait_for_empty_queue(queue, timeout \\ 30_000) do 
      Helpers.wait_for_true(timeout, fn ->
        assert 0 == queue_count(queue)
      end)
    end 
```

Having to leave a comment makes me feel this could have a function wrapped around it for documentation purposes, since we use it a ton in this test file already.

Reply:
I'm going to leave this as is for the time being. There's still some refactoring to do for these tests, as well as adding some new ones.